brave/brave-core
# [sync-v2 base] Sync: Show an infobar at startup for sync v1 users

The infobar displays a notice that sync v2 must be setup in order to continue syncing.
In this infobar, the user can
- perform an action to begin sync v2 setup, or;
- dismiss the infobar.

## Test Plan

_We want a profile with sync v1 enabled, and then open a browser with sync v2 code to test the infobar shows:_

1. Open **release** version of Brave with a new profile:
[macos]`/Applications/Brave\ Browser.app/Contents/MacOS/Brave\ Browser --user-data-dir=$HOME/Library/Application\ Support/BraveSoftware/5804-sync-v2-infobar-test-1`

2. Turn on brave sync flag in chrome://flags

3. Quit (don't use relaunch button)

4. Run again **release** brave with above user data dir command, and enable sync v1 at `brave://sync`
5. Quit again

6. Launch local / nightly / PR build (with sync v2) with same profile via above command (replacing with nightly), or for local build: `yarn start --user_data_dir_name=5804-sync-v2-infobar-test-1`
7. Setup sync v2 via `brave://settings`
8. Quit
9. Relaunch and _**observe infobar**_

![image](https://user-images.githubusercontent.com/741836/84348162-217d5880-ab69-11ea-950f-4bb63c823d9f.png)
